util/src/main/java/org/folio/metastorage/util/AsyncCodec.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.metastorage.util;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+public class AsyncCodec {
+
```

Comment:
Should have private constructor. Okapi has a utility to test that (to keep code coverage high).